### ISSUE TITLE
v0.8.0 dumps core on RHEL

### ISSUE THREAD

**Issue by prjemian** [nexpy/nexpy#84]
using the eclipse+PyDev debugger, single-stepped to line 114 in `nexpy.gui.mainwindow.MainWindow.__init__()`

which is the call to:
```
self.console = NXRichJupyterWidget(config=self.config, parent=rightpane)
```

where self.config = Config: {}
and rightpane = QtGui.QWidget()

**Comment by prjemian** [comment]
The value of self.config is curious.  Where is this defined?  default value is None

**Comment by rayosborn** [comment]
Jupyter has a system for allowing configurations (prompts, fonts, etc) to be defined in startup files. The config dictionary is defined in the Jupyter classes that NXConsoleApp inherits from. I doubt that's what is causing this. It might be worth checking that 'jupyter qtconsole' works from the command line. It might also be worth updating qtconsole or jupyter_client. If there's an incompatibility, I may need to put some minimum version in.

**Comment by rayosborn** [comment]
I came across a Jupyter notebook Github issue concerning a core dump that was caused by having multiple versions of zmq on the system (see https://github.com/jupyter/notebook/issues/378). We don't actually use zmq because we have an in-process kernel, but I think there is an unnecessary import of zmq hidden within other imports in consoleapp.py. NeXpy seems to run fine without them, so I will post an updated version on Anaconda for you to try soon.

**Comment by prjemian** [comment]
Completely removed all jupyter installations and download packages from my RHEL installation.

conda install jupyter jupyter-client jupyter-console jupyter-core

  module | version
  ---------- | -----------
  jupyter | 1.0.0_py27_1
  jupyter-client | 4.1.1_py27_0
  jupyter-console | 4.1.0_py27_0
  jupyter-core | 4.0.6_py27_0

**Comment by prjemian** [comment]
same core dump

**Comment by rayosborn** [comment]
I am guessing that it's a clash of PyQt imports. We are using the Matplotlib backends.qt_compat module to import PyQt4, but the rich_jupyter_widget module uses its own qtconsole.qt module. On other systems, this has worked okay. Could you try setting the environment variable, QT_API, to 'pyqt'? It might be that jupyter is trying to import PyQt5.

**Comment by jpcastell** [comment]
I'm getting the following bug with 0.8  the build core dumps but using pycharm on the source I get the following 

/usr/bin/python2.7 /home/jcastell/nexpy/src/nexpy/nexpygui.py
Traceback (most recent call last):
  File "/home/jcastell/nexpy/src/nexpy/nexpygui.py", line 30, in <module>
    main()
  File "/home/jcastell/nexpy/src/nexpy/nexpygui.py", line 25, in main
    from nexpy.gui.consoleapp import main
  File "build/bdist.linux-x86_64/egg/nexpy/gui/consoleapp.py", line 32, in <module>
  File "build/bdist.linux-x86_64/egg/nexpy/gui/mainwindow.py", line 38, in <module>
  File "/home/jcastell/.local/lib/python2.7/site-packages/qtconsole/inprocess.py", line 5, in <module>
    from ipykernel.inprocess import (
  File "/home/jcastell/.local/lib/python2.7/site-packages/ipykernel/__init__.py", line 2, in <module>
    from .connect import *
  File "/home/jcastell/.local/lib/python2.7/site-packages/ipykernel/connect.py", line 13, in <module>
    from IPython.paths import get_ipython_dir
ImportError: No module named paths


=====================
system

Linux- Ubuntu 15.10

======================

**Comment by prjemian** [comment]
```
bash $ export QT_API=pyqt
bash $ nexpy &
```

This environment setting allows nexpy to start without dumping core.  The application was able to read and plot all example files in the h5py examples section of the NeXus manual.

**Comment by prjemian** [comment]
until a proper fix is coded, I can add this environment definition to nexpygui.py to my local installation for the purposes of having a working nexpy version on /APSshare

**Comment by rayosborn** [comment]
Thanks for checking that out. I guess the systems where it works don't have PyQt5 installed. I actually do have a conda environment on my Mac where PyQt5 is installed, but it doesn't create a conflict. Apparently, in other systems, it does. There may be a way to automatically set the environment variable once the Matplotlib function has determined whether to use PySide or PyQt4 (since PyQt5 is not supported yet), so that the IPython code follows it.

**Comment by rayosborn** [comment]
@jpcastell, does setting the QT_API variable fix your problem? The error you show above seems to suggest there is some misfit between the jupyter modules and your version of IPython. The paths module is certainly in my distribution. Can you try updating IPython?

**Comment by tacaswell** [comment]
what version of mpl do you have?  The fallback order in some corner cases changed to prefer qt5 when things are not clear in 1.5.0.

**Comment by prjemian** [comment]
On the RHEL system
matplotlib                1.5.1               np110py27_0    defaults

**Comment by prjemian** [comment]
same on linux mint system

**Comment by prjemian** [comment]
Interim fix (for me), is to modify */home/oxygen/JEMIAN/Apps/anaconda/bin/nexpy*, adding the environment definition for QT_API=pyqt, as follows:

```
#!/local/Apps/anaconda/bin/python
if __name__ == '__main__':
    import os
    os.environ['QT_API'] = 'pyqt'
    import sys
    import nexpy.nexpygui

    sys.exit(nexpy.nexpygui.main())
```

just the two lines with the *os* package are new

**Comment by tacaswell** [comment]
If you just do

```
import matplotlib
print(matplotlib.get_backend())
```

what do you get?

The fall-through logic on qt_compat is not fun.

**Comment by prjemian** [comment]
on RHEL
```
jemian@gov .../oxygen18/JEMIAN $ ipython
Python 2.7.11 |Anaconda 2.4.1 (64-bit)| (default, Dec  6 2015, 18:08:32) 
Type "copyright", "credits" or "license" for more information.

IPython 4.1.1 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import matplotlib

In [2]: print(matplotlib.get_backend())
Qt4Agg

In [3]: 
```

**Comment by prjemian** [comment]
same on linux mint

it cores on RHEL but runs OK on mint

**Comment by tacaswell** [comment]
hmm, that is not what I expected.

**Comment by rayosborn** [comment]
I have posted a PR #86, which sets the QT_API environment variable, once matplotlib has selected a PyQt library. I have tested it on my Mac with PyQt5 installed, but it would be good to test it on a linux system that has been experiencing core dumps.

**Comment by jpcastell** [comment]
I just did some testing on windows 10 and the new build is working.  it
still doesn't show the current version number.  (using enthought canopy
python distribution)

On Tue, Feb 23, 2016 at 4:34 PM, Ray Osborn <notifications@github.com>
wrote:

> I have posted a PR #86 <https://github.com/nexpy/nexpy/pull/86>, which
> sets the QT_API environment variable, once matplotlib has selected a PyQt
> library. I have tested it on my Mac with PyQt5 installed, but it would be
> good to test it on a linux system that has been experiencing core dumps.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nexpy/nexpy/issues/84#issuecomment-187747608>.
>



-- 
John-Paul Castellan

mobile phone: +33781159511

**Comment by prjemian** [comment]
I'll test on RHEL late today

**Comment by rayosborn** [comment]
Thanks, @prjemian. I presume that what I did is equivalent to you manually setting the QT_API in advance, but you can never be sure.

**Comment by rayosborn** [comment]
This should have been fixed by PR #86. Please reopen if this problem recurs.
